### PR TITLE
El makarova/query issues

### PR DESCRIPTION
**Pull request by Raubzeug**
The following issues are improved in this PR:
1) Typecheck script in package.json contained an error.
2) /viewer/json/query API-point is going to return errors in new format. This feature was implemented into all components that are using this point. Backward compatibility is kept.

### CODE REVIEW

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+    const renderTitle = () => {
+        let content;
+        if (typeof data === 'string') {
+            content = data;
+        } else {
+            const severity = getSeverity(data?.error?.severity);
+            content = (
+                <React.Fragment>
+                    <IssueSeverity severity={severity} /> <span className={blockWrapper('error-message-text')}>{data?.error?.message}</span>
```
Comment (tsufiev) [inline src/containers/Tenant/QueryEditor/Issues/Issues.tsx:43]: If IssueSeverity tag is closed, it's better to move next span tag to a next line.

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    );
+}
+
+interface IssuesProps {
+    className?: string;
+    issues: IssueType[];
+}
+export function Issues({issues, className}: IssuesProps) {
+    const mostSevereIssue = issues.reduce((result, issue) => {
+        const severity = issue.severity ? issue.severity : 10;
```
Comment (tsufiev) [inline src/containers/Tenant/QueryEditor/Issues/Issues.tsx:74]: severity = 0 is false

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        return result < severity ? result : severity;
```
Comment (tsufiev) [inline src/containers/Tenant/QueryEditor/Issues/Issues.tsx:75]: return Math.min(severity, result);

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    }, issues[0].severity ?? 10);
```
Comment (tsufiev) [inline src/containers/Tenant/QueryEditor/Issues/Issues.tsx:76]: 10 is more than enough here, considering you're scanning through all issues, including issues[0].

File: src/containers/Tenant/Diagnostics/TopShards/TopShards.js [hunk 5]
```diff
@@ -123,7 +123,7 @@ function TopShards({
             return 'No data';
         }
         if (error) {
-            return error.data || error;
+            return error.data?.error?.message || error.data || error;
```
[the remark this answers is not in the document]
Comment (Raubzeug) [reply src/containers/Tenant/Diagnostics/TopShards/TopShards.js:126]: Good idea!

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx (hunk 1 above)
Comment (Raubzeug) [reply src/containers/Tenant/QueryEditor/Issues/Issues.tsx:43]: Done!

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx (hunk 2 above)
Comment (Raubzeug) [reply src/containers/Tenant/QueryEditor/Issues/Issues.tsx:74]: Thanks!

File: src/containers/Tenant/QueryEditor/Issues/Issues.tsx (hunk 3 above)
Comment (Raubzeug) [reply src/containers/Tenant/QueryEditor/Issues/Issues.tsx:75]: Done!

**Review by tsufiev — APPROVED** [review]

**Pull request merged**